### PR TITLE
fix changeset parser fail when summary is empty

### PR DESCRIPTION
**Pull request by akphi**
Fix https://github.com/changesets/action/issues/138

@Andarist Could you do me a favor to upgrade `changeset-action` to use the latest parser when we merge this PR? Thank you so much!

### CODE REVIEW

File: packages/parse/src/index.test.ts [hunk 1]
```diff
@@ -148,6 +148,17 @@ describe("parsing a changeset", () => {
       summary: ""
     });
   });
+  it("should be fine if the changeset only contains frontmatter", () => {
```
Comment (Andarist) [inline packages/parse/src/index.test.ts:151]: This is a somewhat different test case that was originally reported. The reported case was about an empty summary but it had some releases declared in the frontmatter. I believe that it's worth adding tests for both of those cases.

File: packages/parse/src/index.test.ts [hunk 2]
```diff
@@ -148,6 +148,17 @@ describe("parsing a changeset", () => {
...
+    const changesetMd = outdent`---
+    ---
+    `;
+
+    const changeset = parse(changesetMd);
```
Comment (Andarist) [inline packages/parse/src/index.test.ts:156]: `outdent` removes a trailing newline by default and all but it feels a little bit implicit here. I would prefer if we could do this explicitly here:
```suggestion
    const changeset = parse('---\n---');
```

File: packages/parse/src/index.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import yaml from "js-yaml";
 import { Release, VersionType } from "@changesets/types";
 
-const mdRegex = /\s*---([^]*?)\n\s*---\s*\n([^]*)/;
+const mdRegex = /\s*---([^]*?)\n\s*---(?:\s*\n([^]*))?$/;
```
Comment (Andarist) [inline packages/parse/src/index.ts:4]: This version of the regexp doesn't allow for a valid~ input like `"---\n--- "`

We can fix this with:
```suggestion
const mdRegex = /\s*---([^]*?)\n\s*---(\s*(?:\n|$)[^]*)/;
```

By using this suggestion we can also revert the change in the `summary` declaration as this regexp doesn't use an optional capturing group so it will always end up using an empty string for a `roughSummary` match.

Comment (akphi) [reply packages/parse/src/index.ts]: Very nice catch! This is updated now

File: packages/parse/src/index.test.ts (hunk 1 above)
Comment (akphi) [reply packages/parse/src/index.test.ts]: Done!

File: packages/parse/src/index.test.ts (hunk 2 above)
Comment (akphi) [reply packages/parse/src/index.test.ts]: I totally overlooked that you have `outdent`. Updated!

File: packages/parse/src/index.test.ts [hunk 4]
```diff
@@ -148,6 +170,13 @@ describe("parsing a changeset", () => {
       summary: ""
     });
   });
+  it("should be fine if the changeset only contains frontmatter", () => {
```
Comment (Andarist) [inline packages/parse/src/index.test.ts:173]:
```suggestion
  it("should be fine if the changeset is empty and without any trailing whitespace", () => {
```

File: packages/parse/src/index.test.ts [hunk 5]
```diff
@@ -136,6 +136,28 @@ describe("parsing a changeset", () => {
...
+    "cool-package": minor
+    ---`;
+
+    const changeset = parse(changesetMd);
+    expect(changeset).toEqual({
+      releases: [{ name: "cool-package", type: "minor" }],
+      summary: ""
+    });
+  });
+  it("should be fine if the summary body contains only an empty space", () => {
```
Comment (Andarist) [inline packages/parse/src/index.test.ts:150]:
```suggestion
  it("should be fine if there is no summary body and the frontmatter has some trailing whitespace", () => {
```

File: packages/parse/src/index.test.ts [hunk 6]
```diff
@@ -136,6 +136,28 @@ describe("parsing a changeset", () => {
       summary: expectedSummary
     });
   });
+  it("should be fine if the summary body is empty", () => {
```
Comment (Andarist) [inline packages/parse/src/index.test.ts:139]:
```suggestion
  it("should be fine if the summary body is completely empty and there is no trailing whitespace", () => {
```

**Comment by Andarist** [comment]
@akphi could you also add a changeset for this change?

> @Andarist Could you do me a favor to upgrade changeset-action to use the latest parser when we merge this PR? Thank you so much!

Sure thing.

**Comment by akphi** [comment]
> @akphi could you also add a changeset for this change?

Done!

File: .changeset/light-cats-flow.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+---
+"@changesets/parse": patch
```
Comment (Andarist) [inline .changeset/light-cats-flow.md:2]:
```suggestion
"@changesets/cli": patch
"@changesets/parse": patch
```
